monai/networks/blocks/split.py

```diff
@@ -0,0 +1,192 @@
+
+
+class SplitConvolution(nn.Module):
+    def __init__(
+        self,
+        spatial_dims: int,
+        in_channels: int,
+        out_channels: int,
+        strides: Sequence[int] | int = 1,
+        kernel_size: Sequence[int] | int = 3,
+        adn_ordering: str = "NDA",
+        act: tuple | str | None = "PRELU",
+        norm: tuple | str | None = "INSTANCE",
+        dropout: tuple | str | float | None = None,
+        dropout_dim: int | None = 1,
+        dilation: Sequence[int] | int = 1,
+        groups: int = 1,
+        bias: bool = True,
+        conv_only: bool = False,
+        is_transposed: bool = False,
+        padding: Sequence[int] | int | None = None,
+        output_padding: Sequence[int] | int | None = None,
+    ) -> None:
+        super().__init__()
+        self.conv = monai.networks.blocks.convolutions.Convolution(
+            spatial_dims,
+            in_channels,
+            out_channels,
+            strides,
+            kernel_size,
+            adn_ordering,
+            act,
+            norm,
+            dropout,
+            dropout_dim,
+            dilation,
+            groups,
+            bias,
+            conv_only,
+            is_transposed,
+            padding,
+            output_padding,
+        )
+
+        self.tp_dim = 1
+
+    def forward(self, x):
+        num_splits = NUM_SPLITS
+        print("num_splits:", num_splits)
+        l = x.size(self.tp_dim + 2)
+        split_size = l // num_splits
+
+        if False:
+            splits = [x[:, :, i * split_size : (i + 1) * split_size, :, :] for i in range(num_splits)]
+        else:
+            padding = SPLIT_PADDING
+            print("padding:", padding)
+
+            overlaps = [0] + [padding] * (num_splits - 1)
+            if self.tp_dim == 0:
+                splits = [
+                    x[
+                        :,
+                        :,
+                        i * split_size - overlaps[i] : (i + 1) * split_size + (padding if i != num_splits - 1 else 0),
+                        :,
+                        :,
+                    ]
+                    for i in range(num_splits)
+                ]
+            elif self.tp_dim == 1:
+                splits = [
+                    x[
+                        :,
+                        :,
+                        :,
+                        i * split_size - overlaps[i] : (i + 1) * split_size + (padding if i != num_splits - 1 else 0),
+                        :,
+                    ]
+                    for i in range(num_splits)
+                ]
+            elif self.tp_dim == 2:
+                splits = [
+                    x[
+                        :,
+                        :,
+                        :,
+                        :,
+                        i * split_size - overlaps[i] : (i + 1) * split_size + (padding if i != num_splits - 1 else 0),
+                    ]
+                    for i in range(num_splits)
+                ]
```

Comment:
```suggestion
            slices=(slice(None),)*(2+self.tp_dim)
            splits=[]
            for i in range(num_splits):
                last_slice=slice( i * split_size - overlaps[i] , (i + 1) * split_size + (padding if i != num_splits - 1 else 0))
                splits.append(x[slices+(last_slice,)])
```

I think this is equivalent? I didn't run it however.